lyndapham/step/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,18 +50,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = datastore.prepare(query);
 
     List<Vote> votes = new ArrayList<>();
+    boolean hidden = false;
+    VoteResponse res;
+
     for (Entity entity : results.asIterable()) {
       String vote = (String) entity.getProperty("vote");
       String comment = (String) entity.getProperty("comment");
 
-      Vote entry = new Vote(vote, comment);
+      double score = (double) entity.getProperty("sentimentScore");
+      Vote entry = new Vote(vote, comment, score);
+      if (score < -0.6) {
+        hidden = true;
+        continue;
+      }
       votes.add(entry);
     }
 
+    if (hidden) {
+      res =
+          new VoteResponse(
+              votes, "Some messages have been hidden due to language! Please be kind.");
+    } else {
+      res = new VoteResponse(votes);
+    }
     Gson gson = new Gson();
-
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(votes));
+    if (!res.getList().isEmpty()) {
```

Comment:
why not just toJson the whole res?

Reply:
It includes the message in the Json format, so it shows "message": "...blah..." which I don't really like.